CRM/Upgrade/Incremental/php/FiveTwentyEight.php

```diff
@@ -26,9 +26,15 @@ class CRM_Upgrade_Incremental_php_FiveTwentyEight extends CRM_Upgrade_Incrementa
    */
   public function setPreUpgradeMessage(&$preUpgradeMessage, $rev, $currentVer = NULL) {
     // Example: Generate a pre-upgrade message.
-    // if ($rev == '5.12.34') {
-    //   $preUpgradeMessage .= '<p>' . ts('A new permission, "%1", has been added. This permission is now used to control access to the Manage Tags screen.', array(1 => ts('manage tags'))) . '</p>';
-    // }
+    if (!function_exists('civi_wp')) {
+      //exit
+    }
+    elseif ($rev == '5.28.alpha1') {
+      $preUpgradeMessage .= '<p>' . ts('Starting with version 5.29.0, CiviCRM on WP will
+        automate the determination of the civicrm files directory. If you have a non-standard
+        directory structure you will need to either override the settings in civicrm.settings.php
```

Comment:
```suggestion
        automate the determination of the civicrm files directory. If you have a legacy (wp-content/plugins/files/civicrm) or non-standard directory structure you will need to either override the settings in civicrm.settings.php
```

Reply:
Ugh.  This one didn't do right and will wipe out the capitalization from above.  Git's a whole different animal.